### ISSUE TITLE
Cannot connect server without passwd and key

### ISSUE THREAD

**Issue by zhaochengwei** [paramiko/paramiko#1418]
If I connect a server without passwd and key， I can not connect the server

![image](https://user-images.githubusercontent.com/13081827/55215338-681cb680-5234-11e9-8ab4-456130927555.png)

**Comment by MrMino** [comment]
The `NoValidConnectionError` means that python cannot create a TCP connection to `"192.168.2.1"` on port `22`. It doesn't mean you cannot authenticate without password or key (which you probably can't anyway, you are just using an ssh agent).

**Comment by zhaochengwei** [comment]
> The `NoValidConnectionError` means that python cannot create a TCP connection to `"192.168.2.1"` on port `22` - maybe the server is dead? It doesn't mean you cannot authenticate without password or key (which you probably can't anyway, you are just using an ssh agent).

![image](https://user-images.githubusercontent.com/13081827/55215311-54715000-5234-11e9-9517-9cc0330c18c9.png)
yes,you are right.  This is the real problem

**Comment by zhaochengwei** [comment]
> The `NoValidConnectionError` means that python cannot create a TCP connection to `"192.168.2.1"` on port `22` - maybe the server is dead? It doesn't mean you cannot authenticate without password or key (which you probably can't anyway, you are just using an ssh agent).
I still can not solve the problem

**Comment by hughesxu** [comment]
If you want to connect to a server with passwordless, you could simply do like:

```
self.client = paramiko.SSHClient()
self.client.set_missing_host_key_policy(paramiko.AutoAddPolicy())
self.client.connect(hostname=self.host, username=self.username, port=self.port, password=None, timeout=self.timeout, allow_agent=False, look_for_keys=False)
```

Note:  Need to add the fix mzakharo/paramiko@6e979ea in code

**Comment by zhaochengwei** [comment]
> If you want to connect to a server with passwordless, you could simply do like:
> 
> ```
> self.client = paramiko.SSHClient()
> self.client.set_missing_host_key_policy(paramiko.AutoAddPolicy())
> self.client.connect(hostname=self.host, username=self.username, port=self.port, password=None, timeout=self.timeout, allow_agent=False, look_for_keys=False)
> ```
> Note: Need to add the fix [mzakharo/paramiko@6e979ea](https://github.com/mzakharo/paramiko/commit/6e979ea) in code

**Comment by zhaochengwei** [comment]
！

> If you want to connect to a server with passwordless, you could simply do like:
> 
> ```
> self.client = paramiko.SSHClient()
> self.client.set_missing_host_key_policy(paramiko.AutoAddPolicy())
> self.client.connect(hostname=self.host, username=self.username, port=self.port, password=None, timeout=self.timeout, allow_agent=False, look_for_keys=False)
> ```
> Note: Need to add the fix [mzakharo/paramiko@6e979ea](https://github.com/mzakharo/paramiko/commit/6e979ea) in code

Thank You very much !!!
